File: src/Shared/Logistics.Shared.Models/Tenant/TenantDto.cs
using System.Text.Json.Serialization;

using Logistics.Domain.Primitives.ValueObjects;

namespace Logistics.Shared.Models;

public record TenantDto
{
    public Guid? Id { get; set; }
    public string? Name { get; set; }
    public string? CompanyName { get; set; }
    public string? BillingEmail { get; set; }
    public string? DotNumber { get; set; }
    public Address? CompanyAddress { get; set; }
    public string? StripeCustomerId { get; set; }

    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
    public string? ConnectionString { get; set; }

    public SubscriptionDto? Subscription { get; set; }

    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public int? EmployeeCount { get; set; }
}

File: src/Client/Logistics.OfficeApp/src/app/core/api/models/tenant.dto.ts
import {AddressDto} from "./address.dto";
import {SubscriptionDto} from "./subscription";

export interface TenantDto {
  id: string;
  name: string;
  companyName: string;
  dotNumber?: string;
  companyAddress: AddressDto;
  subscription: SubscriptionDto | null;
  employeeCount?: number;
}
